### PR TITLE
Fixed tests on Windows systems

### PR DESCRIPTION
**Pull request by javiereguiluz**
Trying to fix the AppVeyor issues by reusing Symfony's `appveyor.yml` file.

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -1,29 +1,59 @@
 build: false
-platform: x86
+clone_depth: 1
 clone_folder: c:\projects\symfony-demo
 
 cache:
-  - '%LOCALAPPDATA%\Composer\files'
+    - c:\projects\symfony\composer.phar
```
Comment (stof) [inline appveyor.yml]: path should be adapter

File: appveyor.yml [hunk 2]
```diff
@@ -1,29 +1,59 @@
...
+    - echo opcache.enable_cli=1 >> php.ini-max
+    - echo extension=php_openssl.dll >> php.ini-max
+    - echo extension=php_apcu.dll >> php.ini-max
+    - echo apc.enable_cli=1 >> php.ini-max
+    - echo extension=php_intl.dll >> php.ini-max
+    - echo extension=php_mbstring.dll >> php.ini-max
+    - echo extension=php_fileinfo.dll >> php.ini-max
+    - echo extension=php_pdo_sqlite.dll >> php.ini-max
+    - echo extension=php_curl.dll >> php.ini-max
+    - echo curl.cainfo=c:\php\cacert.pem >> php.ini-max
```
Comment (stof) [inline appveyor.yml]: you don't need the distinction between ini-max and ini-min for the demo

File: appveyor.yml [hunk 3]
```diff
@@ -1,29 +1,59 @@
...
+    - IF %APPVEYOR_REPO_BRANCH%==master (SET COMPOSER_ROOT_VERSION=dev-master) ELSE (SET COMPOSER_ROOT_VERSION=%APPVEYOR_REPO_BRANCH%.x-dev)
+    - php composer.phar update --no-progress --ansi
+    - SET COMPOSER_ROOT_VERSION=
 
 test_script:
-  - cd %APPVEYOR_BUILD_FOLDER%
-  - php ./vendor/bin/phpunit --verbose
+    - cd c:\projects\symfony-demo
+    - SET X=0
+    - copy /Y c:\php\php.ini-min c:\php\php.ini
```
Comment (stof) [inline appveyor.yml]: no need to test twice with different php.ini

File: appveyor.yml [hunk 4]
```diff
@@ -1,29 +1,53 @@
 build: false
-platform: x86
+clone_depth: 1
 clone_folder: c:\projects\symfony-demo
 
 cache:
-  - '%LOCALAPPDATA%\Composer\files'
```
Comment (stof) [inline appveyor.yml]: keeping the composer cache would still be a good idea though (Symfony does not do it due to the magic it does regarding subtree packages built on the fly, but you can do it)

File: appveyor.yml [hunk 5]
```diff
@@ -1,29 +1,53 @@
...
+    - c:\projects\symfony-demo\composer.phar
+    - .phpunit -> phpunit
```
Comment (stof) [inline appveyor.yml]: this one is useless too, as you don't have such special phpunit install

File: appveyor.yml [hunk 6]
```diff
@@ -1,29 +1,53 @@
...
+    - php composer.phar update --no-progress --ansi
+    - SET COMPOSER_ROOT_VERSION=
 
 test_script:
-  - cd %APPVEYOR_BUILD_FOLDER%
-  - php ./vendor/bin/phpunit --verbose
+    - cd c:\projects\symfony-demo
+    - SET X=0
+    - php vendor/bin/phpunit || SET X=!errorlevel!
+    - exit %X%
```
Comment (stof) [inline appveyor.yml]: you can remove all the magic regarding ``X`` here, as you don't need to run several test commands

**Comment by javiereguiluz** [comment]
@stof thanks for your detailed review! This file is getting really polished 😁

File: appveyor.yml [hunk 7]
```diff
@@ -1,29 +1,51 @@
...
+    - php composer.phar self-update
+    - IF %APPVEYOR_REPO_BRANCH%==master (SET COMPOSER_ROOT_VERSION=dev-master) ELSE (SET COMPOSER_ROOT_VERSION=%APPVEYOR_REPO_BRANCH%.x-dev)
+    - php composer.phar update --no-progress --ansi
+    - SET COMPOSER_ROOT_VERSION=
 
 test_script:
-  - cd %APPVEYOR_BUILD_FOLDER%
-  - php ./vendor/bin/phpunit --verbose
+    - cd c:\projects\symfony-demo
+    - php vendor/bin/phpunit || SET X=!errorlevel!
```
Comment (stof) [inline appveyor.yml]: you forgot to remove the ``|| SET X=!errorlevel!`` part (which is a good way to ensure that the build stay green 😄 as you turn failures into success currently)

**Pull request closed without merging**